src/components/generators/jsonforms-uischemagenerator.ts

```diff
@@ -61,44 +62,48 @@ export class UISchemaGenerator implements IUISchemaGenerator {
      * @param uiSchema The ui schema to wrap in a vertical layout.
      * @returns the wrapped uiSchema.
      */
-    private static wrapInLayoutIfNecessary(uiSchema: IUISchemaElement): ILayout {
-        if (uiSchema.type !== 'VerticalLayout') {
-            let verticalLayout: IVerticalLayout = UISchemaGenerator.createVerticalLayout();
+    private static wrapInLayoutIfNecessary
+        (uiSchema: IUISchemaElement, layoutType: string): ILayout {
+        if (uiSchema['elements'] === undefined) {
+            let verticalLayout: ILayout = UISchemaGenerator.createLayout(layoutType);
             verticalLayout.elements.push(uiSchema);
             return verticalLayout;
         }
         return <ILayout>uiSchema;
     }
 
-    generateDefaultUISchema(jsonSchema: any): IUISchemaElement {
-        let uiSchema = this.generateUISchema(jsonSchema, [], '#', '');
-        return UISchemaGenerator.wrapInLayoutIfNecessary(uiSchema);
+    generateDefaultUISchema(jsonSchema: any, layoutType?: string): IUISchemaElement {
```

Comment:
I think we could use default parameters here instead of an optional and checking for `undefined`, but the interface signature should be fine already (I'm not sure though).